packages/block-library/src/image/edit.js

```diff
@@ -72,23 +72,27 @@ class ImageEdit extends Component {
 
 		this.state = {
 			captionFocused: false,
+			hasError: false,
 		};
 	}
 
 	componentDidMount() {
-		const { attributes, setAttributes } = this.props;
+		const { attributes, setAttributes, noticeOperations } = this.props;
 		const { id, url = '' } = attributes;
-
 		if ( ! id && url.indexOf( 'blob:' ) === 0 ) {
 			const file = getBlobByURL( url );
 
 			if ( file ) {
 				mediaUpload( {
 					filesList: [ file ],
+					allowedType: 'image',
 					onFileChange: ( [ image ] ) => {
 						setAttributes( { ...image } );
 					},
-					allowedType: 'image',
+					onError: ( message ) => {
+						this.setState( { hasError: true } );
```

Comment:
I think if make a call here similar to:
```
						noticeOperations.createErrorNotice( message );
						this.props.setAttributes( {
							url: undefined,
							alt: undefined,
							id: undefined,
							caption: undefined,
						} );
```
We may be able to avoid the need for a state flag.